PeerioTechnologies/peerio-mobile
# Feat to prosemirror

#### Relevant info and issue/PR links 

Mobile will support new message format (rich text) instead of operating on the legacy one (plain text).

The support should work both ways, e.g. messages sent from mobile should be rendered properly as rich text by desktop and vice-versa.

https://app.clubhouse.io/peerio/story/1047/mobile-read-compatibility-for-text-formatting

Mobile should support jumboji:
https://app.clubhouse.io/peerio/story/1046/mobile-render-jumboji

Avatar loading indicator fix:
https://app.clubhouse.io/peerio/story/18427/mobile-loading-indicator-should-match-size-of-loaded-avatar-to-prevent-jumping

#### Testing instructions  

The formatting should be tested both ways - when sending messages from mobile to desktop the formatting should appear correctly on desktop for the recipient and on mobile for the sender, when sending messages from desktop to mobile, the formatting should likewise appear correct for mobile recipient and desktop sender.

the following elements should be supported:

- italic (on mobile produced with single asterisks `*something*`)
- bold (on mobile produced with double asterisks `**something**`)
- strikethrough (on mobile produced with double tildas `~~something~~`)
- emojis (on mobile produced with emoji keyboard or with ascii codes, like - `:), :-(, :-D`)
- links (by typing links)
- hard breaks
- mentions (you should be able to tap mentions to bring up profile view)
- emails (in the form of name@domain.com should be parsed and when tapping email client should open if available)

These elements should work correctly when intertwined with normal text and/or each other. For example a single message can contain bold text, a link and an emoji, and it should still be rendered correctly.

Here are some gifs for guidelines:

![some-emojis](https://user-images.githubusercontent.com/750359/51202170-208cbc80-18cc-11e9-97a0-94b2a6e81037.gif)
![hard-break-support](https://user-images.githubusercontent.com/750359/51202172-208cbc80-18cc-11e9-8d8d-6473bf285b77.gif)
![linkify](https://user-images.githubusercontent.com/750359/51202173-208cbc80-18cc-11e9-9a75-013e902bbfae.gif)
![formatting](https://user-images.githubusercontent.com/750359/51202174-208cbc80-18cc-11e9-9281-4a644344cebd.gif)
![mentions](https://user-images.githubusercontent.com/750359/51202175-208cbc80-18cc-11e9-9b1c-e9529a330863.gif)

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
